src/Domain/Commit.php

```diff
@@ -25,23 +25,23 @@ public static function fromString(string $comment): self
         $hash = '';
         $author = '';
         $date  = '';
-        $comment = [];
+        $commentParts = [];
 
         foreach ($commitMetadata as $line) {
-            if (mb_stripos($line, 'commit ') === 0) {
-                $hash = trim(str_ireplace('commit ', '', $line));
+            if (self::lineContains($line, 'commit')) {
+                $hash = self::getContentFromLineWithout($line, 'commit');
                 continue;
             }
-            if (mb_stripos($line, 'author:',) === 0) {
-                $author = trim(str_ireplace('author:', '', $line));
+            if (self::lineContains($line, 'author:')) {
+                $author = self::getContentFromLineWithout($line, 'author:');
                 continue;
             }
-            if (mb_stripos($line, 'date:') === 0) {
-                $date = trim(str_ireplace('date:', '', $line));
+            if (self::lineContains($line, 'date:')) {
+                $date = self::getContentFromLineWithout($line, 'date:');
                 continue;
             }
```

Comment:
Myślałem przez chwilę czy da radę ukryć jakoś te ifki lub zamienić na switcha czy coś ale chyba się nie da :P

Reply:
Feel free to create PR if you find something about how to refactor it :)